pkg/sql/colexec/timewin/timewin.go

```diff
@@ -123,12 +124,25 @@ func (timeWin *TimeWin) Call(proc *process.Process) (vm.CallResult, error) {
 				result.Status = vm.ExecStop
 				return result, nil
 			}
-			bat, err = result.Batch.Dup(proc.Mp())
-			if err != nil {
-				return result, err
+
```

Comment:
This code is repeat several times.   In theory, you can do the allocation once in Prepare, and never worry about the check in Call.   Just CleanOnlyData and you are good.